cmd/util/cmd/common/flow_client.go

```diff
@@ -47,3 +80,62 @@ func InsecureFlowClient(accessAddress string) (*client.Client, error) {
 
 	return flowClient, nil
 }
+
+// PrepareFlowClientOpts will assemble connection options for the flow client for each access node id
+func PrepareFlowClientOpts(accessNodeIDS []string, insecureAccessAPI bool, snapshot protocol.Snapshot) ([]*FlowClientOpt, error) {
+	flowClientOpts := make([]*FlowClientOpt, 0)
+
+	// convert all IDS to flow.Identifier type, fail early if ID is invalid
+	anIDS := make([]flow.Identifier, 0)
+	for _, anID := range accessNodeIDS {
+		id, err := flow.HexStringToIdentifier(anID)
+		if err != nil {
+			return nil, fmt.Errorf("could not get flow identifer from secured access node id (%s): %w", id, err)
+		}
+
+		anIDS = append(anIDS, id)
+	}
+
+	identities, err := snapshot.Identities(filter.HasNodeID(anIDS...))
+	if err != nil {
+		return nil, fmt.Errorf("failed get identities access node identities (ids=%v) from snapshot: %w", accessNodeIDS, err)
+	}
+	identities = identities.Sort(order.ByReferenceOrder(anIDS))
+
+	// make sure we have identities for all the access node IDs provided
+	if len(identities) != len(accessNodeIDS) {
+		return nil, fmt.Errorf("failed to get identity for all the access node IDs provided: %v, got %v", accessNodeIDS, identities.NodeIDs())
+	}
+
+	// build a FlowClientOpt for each access node identity, these will be used to manage multiple flow client connections
+	for i, identity := range identities {
+		accessAddress := ConvertAccessAddrFromState(identity.Address, insecureAccessAPI)
+
+		// remove the 0x prefix from network public keys
```

Comment:
Consider using [`TrimPrefix`](https://pkg.go.dev/strings#TrimPrefix) instead, as you'll be removing key information and getting it reported as "invalid" or "wrong length" later down the line if the initial characters aren't what you expect.

Reply:
https://github.com/onflow/flow-go/pull/1395/commits/d8c099575b498fb6e219121d0bf2a0910a40217a